MERP_Character_Sheet_BE/Repository/CharacterRepository.cs

```diff
@@ -0,0 +1,94 @@
+﻿using MERP_Character_Sheet_BE.Interfaces;
+using MERP_Character_Sheet_BE.Models;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace MERP_Character_Sheet_BE.Repository
+{
+    public class CharacterRepository: ICharacterRepository
+    {
+        private readonly IServiceScope _scope;
+        private readonly CharacterContext _dbContext;
+
+        public CharacterRepository(IServiceProvider services)
+        {
+            _scope = services.CreateScope();
+            _dbContext = _scope.ServiceProvider.GetRequiredService<CharacterContext>();
+        }
+
+        public async Task<bool> Create(Character _char)
+        {
+            _dbContext.Characters.Add(_char);
+
+            var numberOfItemsCreated = await _dbContext.SaveChangesAsync();
+
+            if (numberOfItemsCreated == 0)
+                return false;
+
+            return true;
+        }
+
+        public async Task<bool> Update(Character _char)
+        {
+            var success = false;
+
+            var existingChar = Get(_char.Id);
+
+            if (existingChar != null)
+            {
+                existingChar.Details.Name = _char.Details.Name;
```

Comment:
There are missing props here, probably its better to create a mapper function for this